src/package.coffee

```diff
@@ -200,7 +200,13 @@ class Package
 
   activateResources: ->
     @activationDisposables = new CompositeDisposable
-    @activationDisposables.add(atom.keymaps.add(keymapPath, map)) for [keymapPath, map] in @keymaps
+
+    @activationDisposables.add atom.config.observe "core.disabledKeymaps", {}, (map) =>
+      value = not _.include(map ? [], @name)
+      if value
+        @activateKeymaps()
+      else if not value
+        @deactivateKeymaps()
```

Comment:
It would be nice to do this config-observing in the `PackageManager`. Performance-wise, it would mean there'd be one observer instead of one for each package, and it would also be more consistent with [the way we observe `disabledPackages`](https://github.com/atom/atom/blob/1d9ecc1d597983c0ac88b3c7f9df1bde3f798c3e/src/package-manager.coffee#L305). I think the code could be very similar to the code for `observeDisabledPackages`.

Would you mind refactoring this code in that way?